kani-compiler/src/codegen_aeneas_llbc/mir_to_ullbc/mod.rs

```diff
@@ -152,31 +187,192 @@ impl<'a, 'tcx> Context<'a, 'tcx> {
         tid.try_into().unwrap()
     }
 
+    /* 
+    fn find_fun_decl_id(&self, def_id: DefId) -> CharonFunDeclId {
+        debug!("register_fun_decl_id: {:?}", def_id);
+        let tid = *self.id_map.get(&def_id).unwrap();
+        debug!("register_fun_decl_id: {:?}", self.id_map);
+        tid.try_into().unwrap()
+    }
+    */
+
+    fn register_type_decl_id(&mut self, def_id: DefId) -> CharonTypeDeclId {
+        debug!("register_type_decl_id: {:?}", def_id);
+        let tid = match self.id_map.get(&def_id) {
+            Some(tid) => *tid,
+            None => {
+                debug!("***Not found!");
+                let tid = CharonAnyTransId::Type(self.translated.type_decls.reserve_slot());
+                self.id_map.insert(def_id, tid);
+                self.translated.all_ids.insert(tid);
+                tid
+            }
+        };
+        debug!("register_type_decl_id: {:?}", self.id_map);
+        tid.try_into().unwrap()
+    }
+
+    fn find_type_decl_id(&self, def_id: DefId) -> CharonTypeDeclId {
```

Comment:
Isn't the previous function (`register_type_decl_id`) sufficient? Or is this used in cases where the decl_id is guaranteed to be created previously?

Reply:
Can you add a comment that explains this?